pandas/tests/groupby/test_transform.py

```diff
@@ -1074,3 +1072,48 @@ def test_transform_lambda_with_datetimetz():
         name="time",
     )
     assert_series_equal(result, expected)
+
+
+def test_transform_cumcount_ngroup():
+    df = DataFrame(dict(a=[0, 0, 0, 1, 1, 1], b=range(6)))
+    g = df.groupby(np.repeat([0, 1], 3))
+
+    # GH 27472
+    result = g.transform("cumcount")
+    expected = g.cumcount()
+    assert_series_equal(result, expected)
+
+    # GH 27468
+    result = g.transform("ngroup")
+    expected = g.ngroup()
+    assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        "backfill",
+        "bfill",
+        "cumcount",
+        "cummax",
+        "cummin",
+        "cumprod",
+        "cumsum",
+        "diff",
+        "ffill",
+        "pad",
+        "pct_change",
+        "rank",
+        "shift",
+        "ngroup",
+    ],
+)
+def test_transformation_kernels_length(func):
+    # This test is to evaluate if after transformation, the index
+    # of transformed data is still the same with original DataFrame
+    # TODO: exceptions are fillna, tshfit and corrwith
```

Comment:
Ideally, these would be xfails in the param list

```python
    pytest.param('fillna', marks=pytest.mark.xfail('reason'),
```

Can you do that here?

Reply:
Sure, nice tip! thanks! Also i found out not all transform functions are tested, e.g. `fillna` or `tshift` etc, is it okay I open another PR to add tests for the untested function? (some might have bug) @TomAugspurger

Reply:
Yep, that would be great.